client/service/src/task_manager.rs

```diff
@@ -80,7 +81,34 @@ impl SpawnTaskHandle {
 			}
 		};
 
-		(self.executor)(Box::pin(future));
+		(self.executor)(Box::pin(future), TaskType::Async);
+	}
+
+	/// Spawns the blocking task with the given name. See also `spawn`.
+	pub fn spawn_blocking(&self, name: &'static str, task: impl Future<Output = ()> + Send + 'static) {
```

Comment:
It would be great to delegate both this function and `spawn` to a common `spawn_inner` function or something, in order to not duplicate the same code as above.